### ISSUE TITLE
Nice-to-have training features

### ISSUE THREAD

**Issue by andreaskoepf** [epfLLM/Megatron-LLM#30]
Nice-to-have feature ideas:

- [x] add effective token/s metric, print in training loop + wandb logging (interesting for instruction tuning & variable sequence length)
- [ ] add option to set the number of checkpoints to keep and delete old ones (often it is enough to keep the one or two last checkpoints)
- [x] allow to specify the wandb run name for finetune.sh as command line arg
- [ ] logging of epoch progress
